### DISCUSSION TITLE
ValidateParams only validates the first mandatory path param?

### DISCUSSION THREAD

**Discussion by ronlange** [jpos/jPOS-EE#265] [Q&A] [answered]
Hi,
I am using a REST transaction manager, and adding some APIs. I stumbled upon the ValidateParams transaction participant while looking at the code and thought it was just what I needed. However, when I started using it, I noticed that even though I have 2 path params, only the first one actually gets verified against the regexp provided in the params list for it.
Looking at the [code](https://github.com/jpos/jPOS-EE/blob/master/modules/qrest/src/main/java/org/jpos/qrest/ValidateParams.java#L187), it seems it starts a for loop, but returns after trying to match the first one, whether it matches or not.
So I figure I'm either missing something, using it incorrectly, or there's a bug.
Which is it? :)

Thanks

**Comment by chhil** [comment]
That appears to be incorrect.

Please open an issue https://github.com/jpos/jPOS-EE/issues and if you can
provide a patch for it that would be great.

The return validParam(ctx, entry, value); needs to be used to continue the
for loop if it is true.


I think the problem exists for all the check methods.

-chhil

**Reply by chhil** [reply]
https://github.com/jpos/jPOS-EE/issues/266

**Reply by ronlange** [reply]
Thanks for opening an issue.
If I have time I'll try to submit a fix.

**Comment by ar** [comment, marked as answer]
This issue is being addressed in #266 and was fixed in #267
